src/ray/gcs/gcs_server/gcs_actor_scheduler.cc

```diff
@@ -71,21 +91,8 @@ void GcsActorScheduler::Schedule(std::shared_ptr<GcsActor> actor) {
   rpc::Address address;
   address.set_raylet_id(node->node_id());
   actor->UpdateAddress(address);
-  // The backend storage is reliable in the future, so the status must be ok.
-  RAY_CHECK_OK(gcs_actor_table_.Put(
-      actor->GetActorID(), actor->GetActorTableData(), [this, actor](Status status) {
-        RAY_CHECK_OK(status);
-        RAY_CHECK_OK(gcs_pub_sub_->Publish(ACTOR_CHANNEL, actor->GetActorID().Hex(),
-                                           actor->GetActorTableData().SerializeAsString(),
-                                           nullptr));
-        // There is no promise that the node the
-        // actor tied to is still alive as the
-        // flush is asynchronously, so just
-        // invoke `Schedule` which will lease
-        // worker directly if the node is still
-        // available or select a new one if not.
-        Schedule(actor);
-      }));
+
+  Schedule(actor);
```

Comment:
Now we can put the code for leasing worker (Line 65) right here. And should update DB after leasing a worker.

Reply:
fixed